src/zenml/integrations/bentoml/model_deployers/bentoml_model_deployer.py

```diff
@@ -388,9 +389,8 @@ def _matches_search_criteria(
                 == config.pipeline_step_name
             )
             and (
-                not config.pipeline_run_id
-                or existing_service_config.pipeline_run_id
-                == config.pipeline_run_id
+                not config.run_name
```

Comment:
Just a clarification for me: This now works because of the logic you implemented in the `ServiceConfig` class, where you set the value of `run_name` to the old `pipeline_run_id` value, correct? Otherwise the legacy deployments would not have a value for `run_name`.

Reply:
Yes, that should be the case here, even tho if `run_name` isn't provider here it would still work using other parameters.